src/AppInstallerCLICore/COMContext.cpp

```diff
@@ -51,22 +51,33 @@ namespace AppInstaller::CLI::Execution
         Logging::SetExecutionStage(static_cast<uint32_t>(m_executionStage));
     }
 
-    void COMContext::SetLoggerContext(const std::wstring_view telemetryCorrelationJson, const std::string& caller)
+    void COMContext::SetContextLoggers(const std::wstring_view telemetryCorrelationJson, const std::string& caller)
     {
         m_correlationData = telemetryCorrelationJson;
-        Logging::Telemetry().SetTelemetryCorrelationJson(telemetryCorrelationJson);
-        Logging::Telemetry().SetCaller(caller);
-        Logging::Telemetry().LogStartup(true);
+
+        std::unique_ptr<AppInstaller::ThreadLocalStorage::PreviousThreadGlobals> setThreadGlobalsToPreviousState;
+        setThreadGlobalsToPreviousState.reset(GetThreadGlobalsActive().SetForCurrentThread());
+
+        SetLoggers();
+        GetThreadGlobalsActive().GetTelemetryLogger().SetTelemetryCorrelationJson(telemetryCorrelationJson);
+        GetThreadGlobalsActive().GetTelemetryLogger().SetCaller(caller);
+        GetThreadGlobalsActive().GetTelemetryLogger().LogStartup(true);
     }
-    
+
     std::wstring_view COMContext::GetCorrelationJson()
     {
         return m_correlationData;
     }
 
+    void COMContext::SetGlobalLoggers()
+    {
+        SetLoggers();
+        AppInstaller::Logging::Telemetry().Initialize();

```

Comment:
We discussed having the global `Initialize` happen naturally as part of the TLS fallback.  What was the issue with that?

Without that change, there are missing calls to `Initialize` on the global object for the other binaries.

Reply:
The issue was that the TraceLogger write is accessing Telemetry object to log activityId as part of AICLI_LOGs which is causing an indefinitely recursion due to initialize being part of the global telemetry object's constructor. But as proposed in our call, I will instead not log activityId in TraceLogger and let it be a problem to solve for future. For now, I am making changes as per your comment.